Foundation/NSURLSession/http/EasyHandle.swift

```diff
@@ -660,3 +639,35 @@ internal extension CFURLSessionEasyCode {
         throw self
     }
 }
+
+#if os(Android)
+extension URLSession {
+    fileprivate static var _sslCertificateAuthorityFilePathPtr: UnsafeMutablePointer<Int8>?
+    /// You can bypass SSL peer verification altogether (not advised)
+    public static var verifyPeerSSLCertificate = true
+    /// See https://curl.haxx.se/docs/sslcerts.html
+    /// For SSL to work you need a "cacert.pem" to be accessible at the path
+    /// provided by setting URLSession.sslCertificateAuthorityFile to a URL.
+    /// Downloadable here: https://curl.haxx.se/docs/caextract.html
+    public static var sslCertificateAuthorityFile: URL {
+        set(value) {
+            verifyPeerSSLCertificate = true
```

Comment:
OK, final offer. I’ve updated the code to ignore the value of verifyPeerSSLCertificate if a CA cert file has been set. This removes the dependency on the order of calls with the presence of a cert file taking preceedence between the two properties which is simpler to reason.

```swift

#if os(Android)
extension URLSession {
    fileprivate static var _sslCertificateAuthorityFilePathPtr: UnsafeMutablePointer<Int8>?
    /// A Boolean that is `true` if peer SSL certificates are to be verified.
    ///
    /// You can bypass verification by setting this property to `false` (not recommended).
    public static var verifyPeerSSLCertificate = true
    /// The URL representing a PEM-encoded certificate authority file to be used for
    /// peer SSL certificate verification. Setting a certificate authority file takes
    /// precedence over tha value of verifyPeerSSLCertificate.
    ///
    /// See:
    /// - https://curl.haxx.se/docs/sslcerts.html
    /// - https://curl.haxx.se/docs/caextract.html
    public static var sslCertificateAuthorityFile: URL? {
        set(value) {
            free(_sslCertificateAuthorityFilePathPtr)
            guard let value = value else {
                _sslCertificateAuthorityFilePathPtr = nil
                return
            }
            guard value.isFileURL && FileManager.default.fileExists(atPath: value.path) else {
                fatalError("'sslCertificateAuthorityFile' does not reference a file that exists")
            }
            value.path.withCString {
                _sslCertificateAuthorityFilePathPtr = strdup($0)
            }
        }
        get {
            if let value = _sslCertificateAuthorityFilePathPtr {
                return URL(fileURLWithPath: String(cString: value))
            }
            else {
                return nil
            }
        }
    }
}
#endif
```